utils/monitor.js

```diff
@@ -0,0 +1,81 @@
+const { NotFound } = require("http-errors");
+const fireStore = require("./firestore");
+const trackedProgressesCollection = fireStore.collection("trackedProgresses");
+const { RESPONSE_MESSAGES } = require("../constants/monitor");
+const { RESOURCE_NOT_FOUND } = RESPONSE_MESSAGES;
+
+/**
+ * Builds a Firestore query to check if a document exists based on the provided query parameters.
+ *
+ * @param {Object} queryParams - The query parameters for checking if a document exists.
+ * @param {string} queryParams.userId - The ID of the user (optional).
+ * @param {string} queryParams.taskId - The ID of the task (optional).
+ * @returns {Firestore.Query} - A Firestore query to check if a document exists.
+ */
+const buildQueryToCheckIfDocExists = (queryParams) => {
```

Comment:
what does this function do? does it check for valid user and task id?

Reply:
it simply builds a Firestore query to fetch the documents based on taskId or userId. the description needs to be updated. Thanks for pointing this out will update it.